crates/sui-core/src/authority/authority_store_pruner.rs

```diff
@@ -548,6 +549,27 @@ impl AuthorityStorePruner {
         Ok(())
     }
 
+    #[cfg(tidehunter)]
+    fn prune_th(
+        perpetual_db: &Arc<AuthorityPerpetualTables>,
+        checkpoint_store: &Arc<CheckpointStore>,
+        num_epochs_to_retain: u64,
+        pruning_watermark: Arc<AtomicU64>,
+    ) -> anyhow::Result<()> {
+        use std::sync::atomic::Ordering;
+        let epoch_id = checkpoint_store
+            .get_highest_executed_checkpoint()?
+            .map(|c| c.epoch)
+            .unwrap_or_default();
+        if epoch_id >= num_epochs_to_retain {
+            let watermark = epoch_id - num_epochs_to_retain + 1;
+            info!("relocation: setting pruning watermark to {}", watermark);
+            pruning_watermark.store(watermark, Ordering::Relaxed);
+            perpetual_db.objects.db.start_relocation()?;
```

Comment:
Should we put todo here since we need to run pruner for checkpoint db as well?

Also, do I read this code correctly that it will call `start_relocation` on every tick of the timer? (should it do it once per epoch instead?)